### PR TITLE
Change all references of PlayerStatModel to AggregatePlayerStatModel.…

### PR DESCRIPTION
**Pull request by Vexrax**
… Make filtering of playerstats happen on frontend, stop LBs from using PlayerStatModel

### Notes:
- Fully moves the code off of PlayerStatModel. 
- Seems like LBs are a bit broken (before my changes) will make a fix for it after these changes

### CODE REVIEW

File: front-end/src/components/player-page/stats.tsx [hunk 1]
```diff
@@ -25,20 +26,27 @@ interface PlayerPageStatsProps {
...
   } = props;
+
+  // This page does not care about a per champion breakdown right now, so merge the data so its by role.
+  // This page also doesnt care about a per team breakdown right now, so merge the team data away
+  let playerStats = deepCopy(props.playerStats); // DeepCopy is needed for some react state BS
+  if (roleConfig) {
+    playerStats = mergePlayerStats(seasonConfig.seasonId, roleConfig?.roleId, playerStats);
+  }
+
   let wins = 0;
```
Comment (Vexrax) [inline front-end/src/components/player-page/stats.tsx:50]: Real change here

File: front-end/src/components/player-page/stats.tsx [hunk 2]
```diff
@@ -72,6 +80,20 @@ export default function PlayerPageStats(props: PlayerPageStatsProps) {
...
+    const key = `${seasonId}-${roleId}`;
+    if(mergedPlayerStats.has(key)) {
+      mergedPlayerStats.set(key, combine(playerStat, mergedPlayerStats.get(key) as AggregatedPlayerStatModel));
+    } else {
+      mergedPlayerStats.set(key, playerStat);
+    }
+  }
+
+  return Array.from(mergedPlayerStats.values());
+}
```
Comment (Vexrax) [inline front-end/src/components/player-page/stats.tsx:99]: Real change here

File: front-end/src/pages/player/player.tsx [hunk 3]
```diff
@@ -123,7 +124,10 @@ function PlayerPage()
     }
     try {
       let numberSeasonId = seasonId === 'RISEN' ? undefined : Number(seasonId);
-      const stats = await GetPlayerStats(profile.overview.puuid, numberSeasonId, roleId, seasonId === 'RISEN');
+      let teamId = undefined; // TODO hook this in to some component
+      let championId = undefined; // TODO hook this in to some component
+      const stats = await GetPlayerStats(profile.overview.puuid, numberSeasonId, roleId, teamId, championId, seasonId === 'RISEN');
+
```
Comment (Vexrax) [inline front-end/src/pages/player/player.tsx:130]: Real change here

File: BackEnd/src/db/leaderboards.ts [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+import { FindManyOptions, MoreThanOrEqual } from 'typeorm';
 import { combine } from '../../../Common/utils';
 import DenylistModel from '../../../Common/models/denylist.model';
+import AggregatedPlayerStatModel from '../../../Common/models/aggregatedplayerstat.model';
 
 const minNumberOfGames = 4;
 
-export async function GetDbLeaderboards(seasonId?: number, roleId?: GameRoles, risenOnly?: boolean, collapseRoles?: boolean): Promise<PlayerStatModel[]> {
+export async function GetDbLeaderboards(seasonId?: number, roleId?: GameRoles, risenOnly?: boolean, collapseRoles?: boolean): Promise<AggregatedPlayerStatModel[]> {
   await ensureConnection();
```
Comment (Vexrax) [inline BackEnd/src/db/leaderboards.ts:12]: Real Change

**Review by afieldi — APPROVED** [review]

**Pull request merged**